src/assembly/collections.ts

```diff
@@ -159,18 +207,25 @@ export function upsert(
       CollectionStatus.Error,
       "Text is empty.",
       "upsert",
-      "",
+      [],
     );
   }
-  const result = hostUpsertToCollection(collection, key, text);
+  const keys: string[] = [];
+  if (key != null) {
+    keys.push(key);
+  }
+
+  const texts: string[] = [text];
+
+  const result = hostUpsertToCollection(collection, keys, texts);
   if (utils.resultIsInvalid(result)) {
     console.error("Error upserting to Text index.");
     return new CollectionMutationResult(
       collection,
       CollectionStatus.Error,
       "Error upserting to Text index.",
       "upsert",
-      "",
+      [],
```

Comment:
All these empty `[]` feel strange.  Why not just make it optional on the constructor instead?

Reply:
i dont want to have to run into some unexpected null issues we can do this later